app/controllers/foreman_ansible/api/v2/hosts_controller_extensions.rb

```diff
@@ -6,31 +6,63 @@ module HostsControllerExtensions
         extend ActiveSupport::Concern
         include ForemanTasks::Triggers
         include ::ForemanAnsible::Concerns::JobInvocationHelper
+        include ::ForemanAnsible::Concerns::ApiCommon
 
         included do
+          def find_resource
+            return true if params[:action] == 'multiple_play_roles'
+            super
+          end
+
+          before_action :find_ansible_roles, :only => [:assign_ansible_roles]
+
           api :POST, '/hosts/:id/play_roles',
               N_('Plays Ansible roles on a host')
-          param :id, String, :required => true
+          param :id, :identifier, :required => true
 
           def play_roles
             composer = job_composer(:ansible_run_host, @host)
             process_response composer.trigger!, composer.job_invocation
           end
 
-          api :POST, '/hosts/play_roles', N_('Plays Ansible roles on hosts')
-          param :id, Array, :required => true
+          api :POST, '/hosts/multiple_play_roles',
+              N_('Plays Ansible roles on hosts')
+          param :host_ids, Array, N_('IDs of hosts to play roles on'),
+                :required => true
 
           def multiple_play_roles
-            composer = job_composer(:ansible_run_host, @host.pluck(:id))
+            host_ids = params.fetch(:host_ids, []).uniq
+            composer = job_composer(:ansible_run_host, host_ids)
             process_response composer.trigger!, composer.job_invocation
           end
+
+          api :GET, '/hosts/:id/ansible_roles',
+              N_('List all Ansible roles for a host')
+          param :id, :identifier, :required => true
+
+          def ansible_roles
+            return unless @host
+            @ansible_roles = @host.all_ansible_roles
+          end
+
+          api :POST, '/hosts/:id/assign_ansible_roles',
```

Comment:
Why this endpoint? It seems to me that we can definitely do this already by sending a POST request to `api/v2/hosts/:id` with the 'ansible_roles' attribute

Reply:
Sure we can, It's just added "from" https://github.com/theforeman/foreman_ansible/pull/81/files#diff-e3a172920758b4851cabcb23ced54a22R64 and also for easier use from hammer. If you think this is redundant, tell me and I'll remove that.